File: JAVA/LibraryTrackerApp.java
import java.util.HashMap;
import java.util.Map;
import java.util.Scanner;

public class LibraryTrackerApp{
public static void main(String[] args){

Scanner input = new Scanner(System.in);

Map<String, String> booksInLibrary = new HashMap<>();
booksInLibrary.put("Chike and the River", "Avaliable");
booksInLibrary.put("The Richest Man In Babylon", "Avaliable");
booksInLibrary.put("The Power Of A Praying Woman", "Avaliable");
booksInLibrary.put("There was a country", "Avaliable");
booksInLibrary.put("The Millionaire FastLane", "Avaliable");
booksInLibrary.put("Rich Dad, Poor Dad", "Avaliable");
booksInLibrary.put("No Excuses", "Avaliable");
booksInLibrary.put("Arrow at Ease", "Avaliable");
booksInLibrary.put("Purple Hibiscus", "Avaliable");
booksInLibrary.put("The Incorruptiple Judge", "Avaliable");
booksInLibrary.put("Chike and the River", "Avaliable");

String menu = """
		1. View all books
		2. Borrow a book
		3. Return a book
		0. Exit
		""";

boolean libraryPortal = true;
while(libraryPortal){
	System.out.println(menu);
	System.out.println("Select an option: ");
	int menuSelect = input.nextInt();
	switch(menuSelect){
		case 1:
			System.out.println(viewAllBooks(booksInLibrary));
			break;
		case 2:
			System.out.println(viewAllBooks(booksInLibrary));
			System.out.println("Select the book you want to borrow: ");
			int bookNumber = input.nextInt();
			System.out.println(borrowAbook(bookNumber, booksInLibrary));
			break;
			
		case 3:
			System.out.println(viewAllBooks(booksInLibrary));
			System.out.println("Select the book you want to borrow: ");
			bookNumber = input.nextInt();
			System.out.println(returnAbook(bookNumber, booksInLibrary));
			break;
		default:
			System.out.println("Invalid selection");
			break;
		case 0:
			libraryPortal = false;
	}
}

}
public static String viewAllBooks(Map<String, String> booksInLibrary){
	int number = 1;
	String result = "";
	for(Map.Entry<String, String> entry : booksInLibrary.entrySet()){
		result += number + ". " + entry.getKey() + ": " + entry.getValue() + "\n";
		number++;
	}
	return result;
}

public static String borrowAbook(int bookNumber, Map<String, String> booksInLibrary){
	int count = 0;
	for(Map.Entry<String, String> entry : booksInLibrary.entrySet()){
		count++;
		if (count == bookNumber){
			booksInLibrary.replace(entry.getKey(), "Borrowed");
		}
	}
	return "Borrowed Successfully";
}

public static String  returnAbook(int bookNumber, Map<String, String> booksInLibrary){
	int count = 0;
	for(Map.Entry<String, String> entry : booksInLibrary.entrySet()){
		count++;
		if (count == bookNumber){
			booksInLibrary.replace(entry.getKey(), "Available");		}
	}
	return "Book returned successfully";
}




}

File: Python/library_tracker_app.py


def view_all_books(books_in_library):
	number = 1;
	result = "";
	for book in books_in_library:
		result += f"{number}. {book}: {books_in_library[book]}\n";
		number += 1
	return result


def borrow_a_book(book_number):
	count = 0
	for book in books_in_library:
		count += 1
		if count == book_number:
			books_in_library[book] = "Borrowed"
			break;
	return books_in_library


def return_a_book(book_number):
	count = 0
	for book in books_in_library:
		count += 1
		if count == book_number:
			books_in_library[book] = "Available";
	return books_in_library

books_in_library = {"Chike and the River": "Avaliable", "Arrow at Ease": "Avaliable", "Purple Hibiscus": "Avaliable", "The Incorruptiple Judge": "Avaliable", 
"The Richest Man In Babylon": "Avaliable", "The Millionaire FastLane": "Avaliable", "Rich Dad, Poor Dad": "Avaliable", "No Excuses": "Avaliable", 
"The Power Of A Praying Woman": "Avaliable", "There was a country": "Avaliable"};

menu = """
	1. View all books
	2. Borrow a book
	3. Return a book
	0. Exit
	"""

library_portal = True
while library_portal:
	print(menu)
	menu_select = int(input("Select an option: "))
	match(menu_select):
		case 1:
			print(view_all_books(books_in_library))
			
		case 2:
			print(view_all_books(books_in_library))
			book_number = int(input("Select the book you want to borrow: "))
			books_in_library = borrow_a_book(book_number)
			number = 1
			for book in books_in_library:
				print(f"{number}. {book}: {books_in_library[book]}")
				number += 1
			
			
		case 3:
			print(view_all_books(books_in_library));
			book_number = int(input("Select the book you want to return: "))
			books_in_library = return_a_book(book_number)
			number = 1
			for book in books_in_library:
				print(f"{number}. {book}: {books_in_library[book]}")
				number += 1
			

		case 0:
			library_portal = False
	























